html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html

```diff
@@ -367,7 +368,240 @@
+                    assert_equals(e.data.message, "some message", "Checking message");
+                    assert_equals(e.data.foo, undefined, "Checking custom property");
+                });
+                t.step(function() {
+                    const error = SyntaxError("some message");
+                    error.foo = "bar";
+                    worker.postMessage(error);
+                });
+            },
+            function() {
+                var t = async_test("TypeError objects can be cloned");
+                t.id = 33;
+                worker.onmessage = t.step_func_done(function(e) {
+                    assert_equals(Object.getPrototypeOf(e.data), TypeError.prototype, "Checking prototype");
+                    assert_equals(e.data.constructor, TypeError, "Checking constructor");
+                    assert_equals(e.data.name, "TypeError", "Checking name");
+                    assert_equals(e.data.message, "some message", "Checking message");
+                    assert_equals(e.data.foo, undefined, "Checking custom property");
+                });
+                t.step(function() {
+                    const error = TypeError("some message");
+                    error.foo = "bar";
+                    worker.postMessage(error);
+                });
+            },
+            function() {
+                var t = async_test("URIError objects can be cloned");
+                t.id = 34;
+                worker.onmessage = t.step_func_done(function(e) {
+                    assert_equals(Object.getPrototypeOf(e.data), URIError.prototype, "Checking prototype");
+                    assert_equals(e.data.constructor, URIError, "Checking constructor");
+                    assert_equals(e.data.name, "URIError", "Checking name");
+                    assert_equals(e.data.message, "some message", "Checking message");
+                    assert_equals(e.data.foo, undefined, "Checking custom property");
+                });
+                t.step(function() {
+                    const error = URIError("some message");
+                    error.foo = "bar";
+                    worker.postMessage(error);
+                });
+            },
+            function() {
+                var t = async_test("URIError objects from other realms are treated as Error");
+                t.id = 35;
+                worker.onmessage = t.step_func_done(function(e) {
+                    assert_equals(Object.getPrototypeOf(e.data), Error.prototype, "Checking prototype");
+                    assert_equals(e.data.constructor, Error, "Checking constructor");
+                    assert_equals(e.data.name, "Error", "Checking name");
+                    assert_equals(e.data.message, "some message", "Checking message");
+                    assert_equals(e.data.foo, undefined, "Checking custom property");
+                });
+                t.step(function() {
+                    const error = frames[0].URIError("some message");
```

Comment:
Can you check Object.getPrototypeof(error), error.constructor and error.name?

Reply:
That shouldn't be necessary in this setup step, I think? But I will add them anyway, and we can always revert.

Reply:
Thanks!